app/controllers/mainsearch.php

```diff
@@ -34,9 +34,12 @@
     $regex['perfect'] = isset($_GET['perfect_match']) ? 'perfect_match=1' : '';
     $regex = array_filter($regex);
     $regex = count($regex) > 0 ? '?' . implode('&', $regex) : '';
+    $scheme = isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] != 'off'
```

Comment:
I think it should be safe to move this to a shared file (e.g. ```/inc/router.php```). Maybe ```transvision_scheme``` as a name?